### PR TITLE
Binder

### LINKED ISSUE #93: Binder config and notebooks
## Description

Binder branch adds a working binder environment but all our notebooks assume there is a running emu server. 
Option 1: build and run emu inside the binder environment
Option 2: use a public URL for a live emu server.

### PR DESCRIPTION
**Pull request by huard**
## Overview

This PR fixes #93 

See https://mybinder.org/v2/gh/bird-house/birdy/binder?filepath=notebooks for testing

### CODE REVIEW

File: notebooks/agu2018_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+       "Output(string='This is just a string', int=7, float=3.14, boolean=True, angle=90.0, time=datetime.time(12, 0), date=datetime.date(2012, 5, 1), datetime=datetime.datetime(2016, 9, 2, 12, 0, tzinfo=tzutc()), string_choice='scissor', string_multiple_choice='gentle albatros', text=<owslib.wps.ComplexDataInput object at 0x7f08db03fdd8>, dataset=<owslib.wps.ComplexDataInput object at 0x7f08db03f6d8>, bbox=None)"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "out = wps.inout()\n",
```
Comment (cehbrecht) [inline notebooks/agu2018_demo.ipynb]: this should be:
```
wps.inout().get()
```

File: notebooks/agu2018_demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "resp = wps.binaryoperatorfornumbers(1, 2, operator='add')\n",
+    "print(resp)\n",
+    "z"
```
Comment (cehbrecht) [inline notebooks/agu2018_demo.ipynb]: remove `z`.

You could add:
```
resp.get().output
```

**Comment by cehbrecht** [comment]
@huard binder setup worked ... but notebook needs fixes ... see comments.

**Comment by tomLandry** [comment]
I'm starting to play with it a couple of minutes ago, making myself confortable. Had about the same comments as Carsten, but I'm reviewing everything from the start using the presentation too.

**Comment by cehbrecht** [comment]
@huard If you work on this PR, give me a sign :)

Otherwise I will make a few changes. I can add another WPS to show an image.

**Comment by cehbrecht** [comment]
... an example with a WPS returning an image can be found here:
https://github.com/bird-house/birdy/blob/master/notebooks/c4cds.ipynb

**Comment by tomLandry** [comment]
Thanks @cehbrecht. It's getting very late to the game for me though. I might go straight to small improvements to the presentation. That's a very tough week. Let's see if @huard is available for these last minute changes.

**Comment by huard** [comment]
I have a 3 hours bus drive that I can dedicate to the presentation.

**Comment by cehbrecht** [comment]
@huard thanks :) I can check things tomorrow European time.

**Comment by huard** [comment]
That's the beauty of international collaboration. Some of our colleagues
wake up at 1h local time !

**Comment by huard** [comment]
Arghh ! I fixed those issues you see but I think I had two windows open and the other one without all the changes got saved... 

You need to type . then TAB

**Comment by huard** [comment]
So the latest version should be a lot better and I've added Carsten's example.
By the way, should get() actually make sure that the process is completed ? Ie wait for the process to complete ?

**Comment by huard** [comment]
https://hub.mybinder.org/user/bird-house-birdy-4bvzfofk/tree/notebooks

**Comment by cehbrecht** [comment]
@huard I had to fix the bash `exports`.

@tomLandry This link works for me to launch binder:
https://mybinder.org/v2/gh/bird-house/birdy/binder?filepath=notebooks

**Comment by cehbrecht** [comment]
@tomLandry binder is getting slower in launching. In the worse case use the nbviewer:
http://nbviewer.jupyter.org/github/bird-house/birdy/blob/binder/notebooks/agu2018_demo.ipynb

**Comment by tomLandry** [comment]
Thanks all. I was able to do video captures and integrate them in the presentation. Presentation is now uploaded and verified in AGU speaker room. I'm presenting in 2 hours, at 9h30 eastern. One last push (well I give another presentation remotely at OGC tech committee this afternoon ...)

**Comment by cehbrecht** [comment]
@huard can we merge to master and update the links?

**Comment by huard** [comment]
Yes, ideally we'd tag the release and reference that in the binder, otherwise the binder gets trashed at every commit.

**Pull request merged**